data/msf/sms.rb

```diff
@@ -0,0 +1,186 @@
+      super
+      add_console_dispatcher(SMSCommandDispatcher)
+    end
+
+    def cleanup
+      self.framework.events.remove_session_subscriber(@inst)
+      remove_console_dispatcher('sms')
+    end
+
+    def name
+      "sms"
+    end
+
+    def desc
+      "Sends a SMS message when recieving a new session through the use of the King Phisher's REST API."
+    end
+
+    class SMSCommandDispatcher < Plugin::SessionSMS
+      include Msf::Ui::Console::CommandDispatcher
+
+      @king_phisher_server = nil
+      @king_phisher_token = nil
+      @sms_number =  nil
+      @sms_carrier =  nil
+
+      def on_session_open(session)
+        print_status("Session received sending SMS...")
+        begin
+          http_client = Rex::Proto::Http::Client.new("#{@king_phisher_server}")
+          http_client.connect
+          request = http_client.request_cgi({
+            'uri'    => '/_/api/sms/send',
+            'query'  => "token=#{@king_phisher_token}&message=Shells+On+Deck!!+Session:+#{session.sid}&phone_number=#{@sms_number}&carrier=#{@sms_carrier}"
+          })
+          response = http_client.send_recv(request)
+        rescue Exception => e
+          print_error("Exception occured, you done goofed!")
+        ensure
+          http_client.close
+        end
+      end
+
+      def name
+        "sms"
+      end
+
+      def read_settings
+        read = nil
+        if File.exist?(Sms_yaml)
+          ldconfig = YAML.load_file("#{Sms_yaml}")
+          @king_phisher_server = ldconfig['king_phisher_server']
+          @king_phisher_token = ldconfig['king_phisher_token']
+          @sms_number = ldconfig['sms_number']
+          @sms_carrier = ldconfig['sms_carrier']
+          read = true
+        else
+          return read
+        end
+        return read
+      end
+
+      def commands
+        {
+          'sms_start'    => "Start SMS alerts for new sessions",
+          'sms_stop'    => "Stop SMS alerts for new sessions",
+          'sms_save'    => "Save SMS settings to #{Sms_yaml}",
+          'sms_set_server'  => "Set domain name of the King Phisher server",
+          'sms_set_token'    => "Set King Phisher's API token",
+          'sms_set_number'  => "Set number to send SMS alerts to on new session",
+          'sms_set_carrier'  => "Set carrier for sending SMS messages",
+          'sms_show_params'  => "Shows currently set or saved parameters"
+        }
+      end
+
+      def cmd_sms_start
+        if read_settings()
+          self.framework.events.add_session_subscriber(self)
+          print_good("Starting SMS plugin, monitoring sessions...")
+        else
+          print_error("Could not read SMS settings!")
+        end
+      end
+
+      def cmd_sms_stop
+        print_good("Stopping SMS alerting!")
+        self.framework.events.remove_session_subscriber(self)
+      end
+
+      def cmd_sms_save
+        if @king_phisher_server and @king_phisher_token and @sms_number and @sms_carrier
```

Comment:
In ruby it's almost always better to use the binary operators ```&&``` and ```||``` instead of the word operators like "and" and "or".